src/Extension/SecurityAdminAccountResetExtension.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace SilverStripe\MFA\Extension;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use SilverStripe\Control\Email\Email;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Extension;
+use SilverStripe\Admin\SecurityAdmin;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\PasswordEncryptor_NotFoundException;
+use SilverStripe\Security\Permission;
+use SilverStripe\Security\SecurityToken;
+
+/**
+ * This extension is applied to SecurityAdmin to provide an additional endpoint
+ * for sending account reset requests.
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property SecurityAdmin $owner
+ */
+class SecurityAdminAccountResetExtension extends Extension
+{
+    private static $allowed_actions = [
+        'reset',
+    ];
+
+    public function reset(HTTPRequest $request): HTTPResponse
+    {
+        if (!$request->isPOST() || !$request->param('ID')) {
+            return $this->owner
+                ->getResponse()
+                ->setStatusCode(400)
+                ->addHeader('Content-Type', 'application/json')
+                ->setBody(json_encode(
+                    [
+                        'error' => _t(__CLASS__ . '.BAD_REQUEST', 'Invalid request')
+                    ]
+                ));
+        }
+
+        $body = json_decode($request->getBody(), true);
+
+        if (!SecurityToken::inst()->check($body['csrf_token'] ?? null)) {
+            return $this->owner
+                ->getResponse()
+                ->setStatusCode(400)
+                ->addHeader('Content-Type', 'application/json')
+                ->setBody(json_encode(
+                    [
+                        'error' => _t(__CLASS__ . '.INVALID_CSRF_TOKEN', 'Invalid or missing CSRF token')
+                    ]
+                ));
+        }
+
+        if (!Permission::check(MemberExtension::MFA_ADMINISTER_REGISTERED_METHODS)) {
+            return $this->owner
+                ->getResponse()
+                ->setStatusCode(403)
+                ->addHeader('Content-Type', 'application/json')
+                ->setBody(json_encode(
+                    [
+                        'error' => _t(
+                            __CLASS__ . '.INSUFFICIENT_PERMISSIONS',
+                            'Insufficient permissions to reset user'
+                        )
+                    ]
+                ));
+        }
```

Comment:
I feel like `jsonResponse()` would be useful here, we've got it in some of the other controllers I think. Would help to reduce some duplicated lines.

Reply:
I've written a trait for this method now, can apply it to other classes as we go.